src/contentChunk.js

```diff
@@ -181,6 +183,24 @@ export const retrievePayload = async fileId => {
   })
 }
 
+export const retrieveBody = async (bodyId, range) => {
+  if (!bodyId) {
+    throw new Error('bodyID not supplied')
+  }
+
+  const fileDetails = await getFileDetails(bodyId)
+
+  const contentEncoding = fileDetails ? (fileDetails.metadata ? fileDetails.metadata['content-encoding'] : null) : null
+  const decompressionStream = getDecompressionStreamByContentEncoding(contentEncoding)
+
+  const bucket = getGridFSBucket()
+  const downloadStream = bucket.openDownloadStream(bodyId, range)
+  downloadStream.on('error', err => { logger.error(err) })
```

Comment:
@rcrichton An error is thrown here with no details when the content byte ranges are invalid (start larger than end byte) Or when in certain combinations that make no sense - i.e start=3 end=100